src/d1to2fix/symbolsearch.d

```diff
@@ -0,0 +1,131 @@
+/******************************************************************************
+
+    Module built on top of dsymbol library facilities which provides means
+    to find D symbols and reasons about them.
+
+    Copyright: Copyright (c) 2016 Sociomantic Labs. All rights reserved
+
+    License: Boost Software License Version 1.0 (see LICENSE for details)
+
+******************************************************************************/
+
+module d1to2fix.symbolsearch;
+
+/******************************************************************************
+
+    Imports
+
+******************************************************************************/
+
+import dsymbol.modulecache;
+import dsymbol.symbol;
+
+/******************************************************************************
+
+    Initializes module cache and all derived facilities by parsing all
+    modules that can be found under specified import paths.
+
+    NB: must be called in main thread before any file conversion workers
+    start to avoid any race conditions
+
+    Params:
+        paths = array of import paths to scan
+
+******************************************************************************/
+
+public void initializeModuleCache(string[] paths)
+{
+    import std.exception;
+    enforce(delegate_cache is null);
+
+    auto module_cache = new ModuleCache(new ASTAllocator);
+    module_cache.addImportPaths(paths);
+
+    const(DSymbol)*[string] delegates;
+
+    void collectNames (const(DSymbol*)[] symbols)
+    {
+        foreach (symbol; symbols)
+        {
+            // dsymbol tracks both delegates and lambdas as "function" thus
+            // next condition may result in false positives but it is ok
+            // as dmd will compile scope lambda declaration just fine
+
+            if (   symbol.kind == CompletionKind.aliasName
+                && symbol.type !is null
+                && symbol.type.name == "function")
+            {
+                delegates[symbol.name] = symbol;
+            }
+
+            // Recursing into aggregates causes stack overflow, probably
+            // a smarter iteration algorithm is needed to do it. Ignoring
+            // them for now and only checking modules/packages:
```

Comment:
Possibly - or something similar with `alias this`. I haven't investigated it as module scope proved enough for dhtnode goals.